lib/driver.js

```diff
@@ -135,6 +135,13 @@ class EspressoDriver extends BaseDriver {
       if (this.opts.app) {
         // find and copy, or download and unzip an app url or path
         this.opts.app = await this.helpers.configureApp(this.opts.app, APP_EXTENSION);
+
+        // Check that it has the INTERNET permission set. This is required to use espresso-server.
+        const adb = await ADB.createADB();
```

Comment:
it looks like a dirty hack. Can we instead move `this.adb` initialisation to a separate method instead of doing it inside `startEspressoSession`. Also, I cannot find where `checkPackagePresent` helper method is defined.

Reply:
It's causing false session failures on CI as well.

I think what I'll do is just catch the error that occurs when the permission is _not_ set and then log a message + error recommending to set it.

Reply:
yep, you can so it if you know which particular error message we need to handle